contracts/multisig-prover/src/execute.rs

```diff
@@ -22,18 +22,30 @@ use crate::{
     types::{BatchID, WorkersInfo},
 };
 
-pub fn construct_proof(deps: DepsMut, message_ids: Vec<String>) -> Result<Response, ContractError> {
-    let key_id = KEY_ID.load(deps.storage)?;
+pub fn construct_proof(
+    deps: DepsMut,
+    env: Env,
+    message_ids: Vec<String>,
+) -> Result<Response, ContractError> {
     let config = CONFIG.load(deps.storage)?;
+    let workers_info = get_workers_info(&deps, &env, &config)?;
+    let new_worker_set = get_next_worker_set(&deps, &env, &config, false)?;
```

Comment:
Can we move these closer to where they are used? Inside the match statement. Makes it easier to read